entur/tiamat
# Use Testcontainers for Integration testing instead of H2

This pull request introduces Testcontainers to replace the H2 in-memory database for integration testing purposes. Testcontainers provides a more robust and realistic testing environment by running actual database containers.

Changes Made

**Database Replacement**: Replaced the H2 in-memory database with Testcontainers.
**Improved Testing Environment**: Utilized Testcontainers to create Dockerized instances of the database for integration tests.
**Realistic Integration Tests**: Ensures that integration tests are conducted in an environment more closely resembling the production database setup.
**Dependencies**: Added necessary dependencies for Testcontainers integration.
**CircleCI**: Updated CircleCI config to run docker test contains
**ActiveMQ: Minor Moved activemq jms changlelog bean from test to main, this will help to run tiamat without google pubsub
**XerceImple**: Updated xercesImpl version from 2.12.0.SP03-RB to 2.12.2. Note 2.12.0.SP03-RB was note an official version  and note available in maven central. This will help external user to compile tiamat without Entur specific dependencies.
How to Test

Ensure Docker is installed on your machine.

Additional Notes

This change enhances the reliability and accuracy of integration tests.
Testcontainers provides a convenient way to manage and use Docker containers during testing.
Some test has been updated and ignored.
Flyway script has been updated. Requires manual update in database to deploy this version
